File: Maximum sum increasing subsequence[23-Oct-2023]/Solution.cpp
// >> CPP CODE <<

class Solution{
		
	public:
	int maxSumIS(int arr[], int n) {
        vector<int> dp(n);

        for (int i = 0; i < n; i++) {
            dp[i] = arr[i];
        }

        for (int i = 1; i < n; i++) {
            int max = 0;
            for (int j = i - 1; j >= 0; j--) {
                if (arr[j] < arr[i]) {
                    max = max>dp[j]?max:dp[j];
                }
            }

            dp[i] += max;
        }

        return *max_element(dp.begin(), dp.end());
    }
};

File: Maximum sum increasing subsequence[23-Oct-2023]/solution.py
#  >> PYTHON CODE <<

class Solution:
	def maxSumIS(self, Arr, n):
		
		dp=[0]*n
		
		for i in range(n): dp[i]=Arr[i]

		for i in range(n):
		    mx=0
		    for j in range(i-1,-1,-1):
		        if(Arr[j]<Arr[i]):
		            mx=max(mx,dp[j])
		    dp[i]+=mx
		
		
		return max(dp)